packages/react/src/use-subscription.ts

```diff
@@ -0,0 +1,15 @@
+import { useState, useEffect, useContext } from 'react';
+import { Subscription } from '@effection/subscription';
+import { EffectionContext } from './context';
+
+export function useSubscription<T>(subscription: Subscription<T>): T | undefined {
+  let scope = useContext(EffectionContext);
+  let [state, setState] = useState<T>();
+
+  useEffect(() => {
```

Comment:
apologies for the driveby comment but a `useEffect` without arguments will run on every render.

There is an eslint plugin that is pretty good for what is missing in the weird world of react hooks and it is so easy to screw up it really is worth having them like this

```json
  "rules": {
    // ...
    "react-hooks/rules-of-hooks": "error", // Checks rules of Hooks
    "react-hooks/exhaustive-deps": "error" // Checks effect dependencies
  }
  ```

Reply:
Great catch! I totally forgot about this.

Reply:
I've added a fix and added the eslint rules